src/typeguard/_checkers.py

```diff
@@ -530,6 +534,23 @@ def check_typevar(
             )
 
 
+if sys.version_info >= (3, 8):
+    if typing_extensions is None:
+
+        def _is_literal_type(typ: object) -> bool:
+            return typ is typing.Literal
+
+    else:
+
+        def _is_literal_type(typ: object) -> bool:
+            return typ is typing.Literal or typ is typing_extensions.Literal
```

Comment:
If you wanted to avoid the `if typing_extensions is None` check here, you could also do:

```suggestion
    def _is_literal_type(typ: object) -> bool:
        return typ is typing.Literal or typ is getattr(typing_extensions, "Literal", object())
```

I'm easy either way